src/compiler/types.ts

```diff
@@ -4228,6 +4228,8 @@ namespace ts {
      * If changing this, remember to change `moduleResolutionIsEqualTo`.
      */
     export interface ResolvedModuleFull extends ResolvedModule {
+        /* @internal */
+        readonly originalPath: string | undefined;

```

Comment:
you would want to add check for originalPath in ```moduleResolutionIsEqualTo``` as well

Reply:
Why is originalPath not just optional? Also why is it in ResolvedModuleFull and not in ResolvedModule?

Reply:
The only reason we have both `ResolvedModule` and `ResolvedModuleFull` is backwards compatibility. Since this is an internal property it doesn't really matter where it goes.

Reply:
Why is it then string|undefined.. Looking at the changes it seems like it could be made optional instead to avoid setting it to undefined in all the other cases except 1